cli/src/benchmark.rs

```diff
@@ -0,0 +1,208 @@
+use std::{
+    env,
+    num::ParseIntError,
+    process::Command,
+    time::{Duration, Instant, SystemTime, UNIX_EPOCH},
+};
+
+use anyhow::Context;
+use chrono::NaiveDateTime;
+use probe_rs::MemoryInterface;
+use probe_rs_cli_util::{clap, common_options::ProbeOptions};
+
+use rand::prelude::*;
+use serde::{Deserialize, Serialize};
+
+const SIZE: usize = 0x1000;
+
+#[derive(clap::Parser)]
+pub struct BenchmarkOptions {
+    /// Start address for the benchmark test.
+    ///
+    /// Should be located in RAM.
+    #[clap(long = "address", value_parser= parse_hex)]
+    address: u64,
+    #[clap(long = "pr")]
+    pr: Option<u64>,
+
+    #[clap(long)]
+    upload: bool,
+}
+
+fn parse_hex(src: &str) -> Result<u64, ParseIntError> {
+    u64::from_str_radix(src.trim_start_matches("0x"), 16)
+}
+
+pub fn benchmark(common_options: ProbeOptions, options: BenchmarkOptions) -> anyhow::Result<()> {
+    let probe = common_options.attach_probe()?;
+
+    let protocol_name = probe
+        .protocol()
+        .map(|p| p.to_string())
+        .unwrap_or("Unknown protocol".to_string());
```

Comment:
Can we fix this right away, even tho it was not introduced by you? :)

Reply:
Not sure what you mean, what do you want to fix? Stop the benchmark if the protocol is not known?

Reply:
The clippy complaint :)